test_conformance/basic/test_fpmath.cpp

```diff
@@ -98,14 +98,14 @@ int verify_fp(std::vector<T> (&input)[2], std::vector<T> &output,
     auto &inB = input[1];
     for (size_t i = 0; i < output.size(); i++)
     {
-        bool nan_test = false;
+        bool nan_test = true;
 
         T r = test.ref(inA[i], inB[i]);
 
         if (std::is_same<T, cl_half>::value)
-            nan_test = !(isHalfNan(r) && isHalfNan(output[i]));
+            nan_test = (isHalfNan(r) != isHalfNan(output[i]));
 
-        if (r != output[i] && nan_test)
+        if (r != output[i] && !nan_test)
```

Comment:
Hi!  This seems like it has the same problem as before.  Since `nan_test` is true for all types other than `half`, it still skips verification for `float` and `double`, correct?

Instead of trying to fix the NaN checks, can we tear it all out completely?  It's curious to me that we have special handling for `half` NaNs, but we don't have any special handling for `float` or `double` NaNs.

Reply:
> This seems like it has the same problem as before. 

😅... Yes, should check the code again. 

> can we tear it all out completely? 

Use std::isnan for float/double NaN handling.